src/components/MainNavBar/ULMSDialog.tsx

```diff
@@ -0,0 +1,257 @@
+import React from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import { Button, TextField } from "@madie/madie-design-system/dist/react";
+import {
+  Dialog,
+  IconButton,
+  DialogActions,
+  Divider,
+  FormHelperText,
+} from "@mui/material";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import { useFormik } from "formik";
+import { APISchemaValidator } from "./APISchemaValidator";
+import axios from "axios";
+import qs from "qs";
+
+export interface UMLSDialogProps {
+  open: boolean;
+  handleClose: Function;
+  saveTGT: Function;
+  handleToast: Function;
+}
+
+export interface KeyConfig {
+  apiKey: String;
+}
+
+const UMLSDialog = (props: UMLSDialogProps) => {
+  const { open, handleClose, handleToast, saveTGT } = props;
+  // blank, close.
+  const onClose = () => {
+    handleClose();
+    formik.resetForm();
+  };
+  /* 
+    /cas/v1/api-key    Retrieves a Ticket Granting Ticket (TGT)
+    /cas/v1/tickets/{TGT}	Retrieves a single-use Service Ticket
+*/
+  const login = async (values: KeyConfig) => {
+    const url = "https://utslogin.nlm.nih.gov/cas/v1/api-key";
```

Comment:
I think this can be re-written using `https://vsac.nlm.nih.gov/vsac/ws` like @RohitKandimalla mentioned in VSAC spike presentation. it is easier. that way you don't have to parse the html.
url is `https://vsac.nlm.nih.gov/vsac/ws/Ticket  ` and `apikey={apikey}` is the request body.

Reply:
This doesn't appear to work.
Regardless of allowing all cross origin headers, it doesn't pass the pre flight.
Do we have to manually allow that endpoint somewhere?